src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -56,18 +56,49 @@ public bool ExecuteCommandWorker(ReturnKeyCommandArgs args)
             var spans = textView.Selection.GetSnapshotSpansOnBuffer(subjectBuffer);
 
             // Don't split strings if there is any actual selection.
-            if (spans.Count == 1 && spans[0].IsEmpty)
+            // We must check all spans to account for multi-carets.
+            if (!spans.IsEmpty() && spans.All(s => s.IsEmpty))
             {
```

Comment:
The if statement becomes too long. It maybe better to inverse the condition for the quick bail out. For readability purposes.